src/main/scala/chisel3/util/circt/DPI.scala

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt.dpi
+
+import chisel3._
+object RawClockedNonVoidFunctionCall {
+
+  /** Creates an intrinsic that calls non-void DPI function at its clock posedge.
+    *
+    * @example {{{
+    * val a = RawClockedNonVoidFunctionCall("dpi_func_foo", UInt(1.W), clock, enable, b, c)
+    * }}}
+    */
+  def apply[T <: Data](functionName: String, ret: => T)(clock: Clock, enable: Bool, data: Data*): Data = {
+    IntrinsicExpr("circt_dpi_call", ret, "functionName" -> functionName, "isClocked" -> 1)(
+      (Seq(clock, enable) ++ data): _*
+    )
+  }
+}
+
+object RawUnlockedNonVoidFunctionCall {
+
+  /** Creates an intrinsic that calls non-void DPI function when input values are changed.
+    *
+    * @example {{{
+    * val a = RawUnlockedNonVoidFunctionCall("dpi_func_foo", UInt(1.W), enable, b, c)
+    * }}}
+    */
+  def apply[T <: Data](functionName: String, ret: => T)(enable: Bool, data: Data*): Data = {
```

Comment:
I have been thinking: how to pass a string to DPI in Chisel, as well as Plusarg, maybe we also need a `StringLit` type in Chisel.

Reply:
Yes. We'd like to pass and create a string, open array or chandle as well (chandle can be representable with UInt<64> though). My plan is to introduce "intrinsic/foreign type" concept so they can be created by intrinsics.

Reply:
But we may need a Chisel String data type.

Reply:
Yes, I think creating Chisel/FIRRTL String type is a right way but it will require more work. I guess we can use `Vec<UInt<8>, n>` as a way to represent string :)

Reply:
mind blowing. But yes, we can do it